fem/eltrans.hpp

```diff
@@ -356,12 +379,57 @@ class IntegrationPointTransformation
    void Transform (const IntegrationRule  &, IntegrationRule  &);
 };
 
-class FaceElementTransformations
+class FaceElementTransformations : public IsoparametricTransformation
 {
+private:
+   int mask;
+
+   IntegrationPoint eip1, eip2;
+
 public:
-   int Elem1No, Elem2No, FaceGeom;
-   ElementTransformation *Elem1, *Elem2, *Face;
+   int Elem1No, Elem2No;
+   Geometry::Type &FaceGeom; ///< @deprecated Use GetGeometryType instead
+   ElementTransformation *Elem1, *Elem2;
+   ElementTransformation *Face; ///< @deprecated No longer necessary
    IntegrationPointTransformation Loc1, Loc2;
+
+   FaceElementTransformations() : FaceGeom(geom), Face(this) {}
```

Comment:
I don't see how `geom` gets a default value from this construction. Probalby the constructor of `ElementTransformation` should set it to -1.

Reply:
Good catch. `ElementTransformation::geom` has never been set by default but I would also prefer if it were.  This construction in `FaceElementTransformation` is only needed to initialize the reference `FaceGeom` for backward compatibility.  This ensures that `FaceGeom` will always access the same value that is stored in `geom` even if `geom` isn't initialized.